### ISSUE TITLE
When pull is in a real device and the screen switches off, an exception is thrown

### ISSUE THREAD

**Issue by ifoche** [EyeSeeTea/malariapp#585]

**Comment by ivanarrizabalaga** [comment]
http://stackoverflow.com/questions/11344333/how-to-prevent-automatic-screen-lock-on-android-by-code

**Comment by idelcano** [comment]
I added in the xml of the progressActivity the attribute:
    android:keepScreenOn="true"
this prevents the application to switches off.

http://developer.android.com/intl/es/training/scheduling/wakelock.html
Important: i only can try in my device (API 21) but i think it works in all versions.

@ifoche I added a control of pull using a shared Prefereces. 

If you do pull, the preference settings called R.string.pull is set to R.string.incompleted_pull
if the pull is completed the  Preference R.string.pull is set to R.String.completed_pull
else is always R.string.incompleted_pull(or not exist), and if you run the aplication is needed do the full pull.

That is not exactly the card, but i think this task is related and necesary.

I found another problem.

If the conexion is lost during the pulling, the application show the error, but you cant do anything more, you don´t know if is working... freeze..
